documentation/List-of-Action-plugins.md

```diff
@@ -142,31 +143,86 @@ This will look for a user record with `username = "js"` and update it with the _
 
 ---
 
-### Generate Name (or any custom string)
+### Generate Text String
 
-Creates a custom name string and applies it to a specified database record (usually `application.name`)
+Generates serial numbers or arbitrary text strings (e.g. for application name) using a specified pattern, and updates a database record if requested.
 
-- _Action Code:_ **`generateName`**
+- _Action Code:_ **`generateTextString`**
 
-| Input parameters<br />(\*required) <br/>  | Output properties |
-| ----------------------------------------- | ----------------- |
-| `formatExpression`                        | `generatedName`   |
-| `fieldName` (default `"name"`)            |                   |
-| `matchField` (default `"id"`)             |                   |
-| `matchValue` (default `applicationId`)    |                   |
-| `shouldUpdateDatabase` (default `true`)   |                   |
-| `additionalData`                          |                   |
-| `fallbackString` (default `"UNRESOLVED"`) |                   |
+| Input parameters<br />(\*required) <br/> | Output properties |
+| ---------------------------------------- | ----------------- |
+| `pattern`\*                              | `generatedText`   |
+| `counterName`                            |                   |
+| `counterInit`                            |                   |
+| `customFields`                           |                   |
+| `numberFormat`                           |                   |
+| `fallbackText`                           |                   |
+| `updateRecord` (default `false`)         |                   |
+| `tableName` (default `application`)      |                   |
+| `fieldName` (default `name`)             |                   |
+| `matchField` (default `"id"`)            |                   |
+| `matchValue` (default `applicationId`)   |                   |
+| `additionalData`                         |                   |
+
+This action uses the [custom_string_patterns](https://www.npmjs.com/package/custom_string_patterns), which allows the generation of random strings using Regular Expressions, but with the added functionality of also including:
+
+- incrementing, persistent counters
+- custom replacement functions
+- object data replacement.
+
+Please see the custom_string_patterns documentation for full detail, but basically the `pattern` field specified the string to be generated using:
```

Comment:
Where is this `custom_string_patterns` documentation? Maybe a link here?

Reply:
The link is just a few lines further up.

I might add it in here again, but I'll re-visit it after addressing your other comments below, since I'll probably restructure the documentation somewhat.